rehlds/engine/net_chan.cpp

```diff
@@ -1389,8 +1389,10 @@ void Netchan_FlushIncoming(netchan_t *chan, int stream)
 {
 	fragbuf_t *p, *n;
 
+#ifndef REHLDS_FIXES
 	SZ_Clear(&net_message);
 	msg_readcount = 0;
+#endif

```

Comment:
I think this is needed to prevent reading messages from the dropped client (i.e. a client was dropped when reading in the middle of the buffer). It would be better to check if we're dropping the current client:
```suggestion
#ifdef REHLDS_FIXES
	if (chan->player_slot == host_client - g_psvs.clients)
#endif
	{
		SZ_Clear(&net_message);
		msg_readcount = 0;
	}
```